freqtrade/strategy/interface.py

```diff
@@ -312,22 +312,30 @@ def should_sell(self, trade: Trade, rate: float, date: datetime, buy: bool,
         experimental = self.config.get('experimental', {})
 
         if buy and experimental.get('ignore_roi_if_buy_signal', False):
-            logger.debug('Buy signal still active - not selling.')
+            logger.debug(f"'{trade.pair}' - Buy signal still active. Not selling "
+                         f"(sell_type=SellType.NONE)...")
             return SellCheckTuple(sell_flag=False, sell_type=SellType.NONE)
 
         # Check if minimal roi has been reached and no longer in buy conditions (avoiding a fee)
         if self.min_roi_reached(trade=trade, current_profit=current_profit, current_time=date):
-            logger.debug('Required profit reached. Selling..')
+            logger.debug(f"'{trade.pair}' - Required profit reached. Selling "
+                         f"(sell_type=SellType.ROI)...")
             return SellCheckTuple(sell_flag=True, sell_type=SellType.ROI)
 
         if experimental.get('sell_profit_only', False):
-            logger.debug('Checking if trade is profitable..')
+            logger.debug(f"'{trade.pair}' - Checking if trade is profitable...")
             if trade.calc_profit(rate=rate) <= 0:
+                logger.debug(f"'{trade.pair}' - Trade is not profitable. Not selling "
+                             f"(sell_type=SellType.NONE)...")
                 return SellCheckTuple(sell_flag=False, sell_type=SellType.NONE)
+
         if sell and not buy and experimental.get('use_sell_signal', False):
-            logger.debug('Sell signal received. Selling..')
+            logger.debug(f"'{trade.pair}' - Sell signal received. Selling "
```

Comment:
I do not see any extra redundant messages:
```
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Trade is not profitable. sell_flag=False
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,134 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,135 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,135 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,135 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Checking if trade is profitable...
2019-09-12 02:55:13,135 - freqtrade.strategy.interface - DEBUG - XMR/BTC - Sell signal received. sell_flag=True, sell_type=SellType.SELL_SIGNAL
2019-09-12 02:55:13,135 - freqtrade.optimize.backtesting - DEBUG - XMR/BTC - Locking pair till close_time=2019-06-01 12:55:00+00:00
2019-09-12 02:55:13,136 - freqtrade.optimize.backtesting - DEBUG - EOS/BTC - Backtesting emulates creation of new trade: Trade(id=None, pair=EOS/BTC, amount=1031246.77735382, open_rate=0.00096970, open_since=2019-06-01 11:30:00 (3 months ago)).
2019-09-12 02:55:13,136 - freqtrade.persistence - DEBUG - EOS/BTC - Assigning new stoploss...
2019-09-12 02:55:13,137 - freqtrade.persistence - DEBUG - EOS/BTC - Stoploss adjusted. current_price=0.00096970, open_rate=0.00096970, max_rate=0.00097480, initial_stop_loss=0.00085307, stop_loss=0.00085307. Trailing stoploss saved us: 0.00000000.
2019-09-12 02:55:13,137 - freqtrade.strategy.interface - DEBUG - EOS/BTC - Checking if trade is profitable...
```

Reply:
well you're thinking about backtesting, where it's correct - i'm thinking about dry/trade runs - where sell_reason is part of the "notify_sell" notification (which is also logged).

Let's leave as is since it positively impacts backtesting :+1: